pallets/frequency-tx-payment/src/lib.rs

```diff
@@ -102,16 +166,62 @@ pub struct ChargeFrqTransactionPayment<T: Config>(#[codec(compact)] BalanceOf<T>
 
 impl<T: Config> ChargeFrqTransactionPayment<T>
 where
-	BalanceOf<T>: Send + Sync + FixedPointOperand,
+	BalanceOf<T>: Send + Sync + FixedPointOperand + IsType<CapacityBalanceOf<T>> + From<u64>,
+	<T as frame_system::Config>::RuntimeCall:
+		Dispatchable<Info = DispatchInfo, PostInfo = PostDispatchInfo> + IsSubType<Call<T>>,
 {
 	/// Utility construct from tip.
 	pub fn from(tip: BalanceOf<T>) -> Self {
 		Self(tip)
 	}
 
 	/// Return the tip as being chosen by the transaction sender.
-	pub fn tip(&self) -> BalanceOf<T> {
-		self.0
+	pub fn tip(&self, call: &<T as frame_system::Config>::RuntimeCall) -> BalanceOf<T> {
+		match call.is_sub_type() {
+			Some(Call::pay_with_capacity { .. }) => Zero::zero(),
+			_ => self.0,
+		}
+	}
+
+	// Note consider setting the tip at the benning
+	fn withdraw_fee(
+		&self,
+		who: &T::AccountId,
+		call: &<T as frame_system::Config>::RuntimeCall,
+		info: &DispatchInfoOf<<T as frame_system::Config>::RuntimeCall>,
+		len: usize,
+	) -> Result<(BalanceOf<T>, InitialPayment<T>), TransactionValidityError> {
+		let fee =
+			pallet_transaction_payment::Pallet::<T>::compute_fee(len as u32, info, self.tip(call));
+
+		match call.is_sub_type() {
+			Some(Call::pay_with_capacity { .. }) => {
+				let msa_id = pallet_msa::Pallet::<T>::ensure_valid_msa_key(who).map_err(
+					|_| -> TransactionValidityError { InvalidTransaction::Payment.into() },
+				)?;
+
+				T::Capacity::withdraw(msa_id, fee.into()).map_err(
```

Comment:
I think we can talk about this more tomorrow. It is worth exploring.

Reply:
I think this may be worth exploring also, but I will approve since I don't think this is blocking

Reply:
(Feel free to disregard as I am not into the weeds and this might be really wrong)

I think all we care is that they have capacity. Currently capacity is restricted to providers, so that is enough without the additional check.

If we were to change the requirements for capacity, it would be on the acquisition of capacity not the use of it. (for faster checks at use time)